# HG changeset patch
# User Pantxo Diribarne <pantxo.diribarne@gmail.com>
# Date 1374187991 -7200
#      Fri Jul 19 00:53:11 2013 +0200
# Node ID 64d603f8bcd9aca791b57350a39e4272b8ff1044
# Parent  942d892524b35171d38353c207ef72446459065a
__get_axis_args.m: search for 'parent' in property/value pairs and use it as fall back axes.

diff --git a/scripts/plot/__plt_get_axis_arg__.m b/scripts/plot/__plt_get_axis_arg__.m
--- a/scripts/plot/__plt_get_axis_arg__.m
+++ b/scripts/plot/__plt_get_axis_arg__.m
@@ -28,16 +28,21 @@ function [h, varargin, narg] = __plt_get
   if (islogical (caller))
     nogca = caller;
     caller = varargin{1};
     varargin(1) = [];
   else
     nogca = false;
   endif
 
+
+  ## Search for parent property
+  charargs = cellfun (@ischar,varargin);
+  isparent = strcmp (lower (varargin(charargs)), "parent");
+  
   ## Figure handles are integers, but object handles are non-integer,
   ## therefore ignore integer scalars.
   if (nargin > 1 && length (varargin) > 0 && isnumeric (varargin{1})
       && numel (varargin{1}) == 1 && ishandle (varargin{1}(1))
       && varargin{1}(1) != 0 && ! isfigure (varargin{1}(1)))
     tmp = varargin{1};
     obj = get (tmp);
     if ((strcmp (obj.type, "axes") && ! strcmp (obj.tag, "legend"))
@@ -45,16 +50,32 @@ function [h, varargin, narg] = __plt_get
       h = ancestor (tmp, "axes");
       varargin(1) = [];
       if (isempty (varargin))
         varargin = {};
       endif
     else
       error ("%s: expecting first argument to be axes handle", caller);
     endif
+  elseif (numel (varargin) > 1 && any (isparent))
+    tmp = find (charargs);
+    idx = tmp(isparent)(1);
+    if (idx < numel (varargin) && ishandle (varargin{idx+1}))
+      tmp = varargin{idx+1};
+      obj = get (tmp);
+      if ((strcmp (obj.type, "axes") && ! strcmp (obj.tag, "legend"))
+          || strcmp (obj.type, "hggroup"))
+        h = ancestor (tmp, "axes");
+        varargin(idx:idx+1) = [];
+      else
+        error ("%s: expecting parent value to be axes handle", caller);
+      endif
+    else
+      error ("%s: expecting parent value to be axes handle", caller);
+    endif
   else
     f = get (0, "currentfigure");
     if (isempty (f))
       h = [];
     else
       h = get (f, "currentaxes");
     endif
     if (isempty (h))
